cosmos/scheduler/main.go

```diff
@@ -53,8 +53,13 @@ func init() {
 func main() {
 	mongoOptions := options.Client()
 	mongoOptions.SetAppName("cosmos-scheduler")
-	mongoOptions.SetReadPreference(readpref.Secondary())
 	mongoOptions.ApplyURI("mongodb://"+viper.GetString("mongodb_hosts"))
+	// incase of a singleton where no secondary is available
+	if len(mongoOptions.Hosts) == 1 {
+		mongoOptions.SetReadPreference(readpref.Nearest())
+	} else {
+		mongoOptions.SetReadPreference(readpref.Secondary())
```

Comment:
My rationale behind this was because this program will never need to write to MongoDB hence the preference on the secondary (although secondaries will be unavailable singleton deployments). This way there is more available connections to the primary for services that do require it.

Reply:
Switched this to the Primary.